contrib/core-contract-tests/tests/pox-4/pox-4.prop.test.ts

```diff
@@ -0,0 +1,1605 @@
+              "can-stack-stx",
+              [
+                Cl.tuple({
+                  version: Cl.buffer(Uint8Array.from([version])),
+                  hashbytes: Cl.buffer(Uint8Array.from(hashbytes)),
+                }),
+                Cl.uint(amount_ustx),
+                Cl.uint(first_rew_cycle),
+                Cl.uint(num_cycles),
+              ],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.ResponseOk));
+            assert(isClarityType(actual.value, ClarityType.BoolTrue));
+            expect(actual).toBeOk(Cl.bool(expectedResponseOk));
+          }
+        )
+      );
+    });
+
+    it("should return (err 13) can-stack-stx for pox addresses having version > 6", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.integer({
+            min: 7,
+            max: 255,
+          }),
+          fc.array(fc.nat({ max: 255 }), {
+            minLength: 32,
+            maxLength: 32,
+          }),
+          fc.bigInt({
+            min: MIN_AMOUNT_USTX,
+            max: MAX_CLAR_UINT,
+          }),
+          fc.nat(),
+          fc.integer({ min: 1, max: 12 }),
+          (
+            caller,
+            version,
+            hashbytes,
+            amount_ustx,
+            first_rew_cycle,
+            num_cycles
+          ) => {
+            // Arrange
+            const { result: pox_4_info } = simnet.callReadOnlyFn(
+              "pox-4",
+              "get-pox-info",
+              [],
+              caller
+            );
+            assert(isClarityType(pox_4_info, ClarityType.ResponseOk));
+            assert(isClarityType(pox_4_info.value, ClarityType.Tuple));
+            const expectedResponseErr = ERR_STACKING_INVALID_POX_ADDRESS;
+            // Act
+            const { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "can-stack-stx",
+              [
+                Cl.tuple({
+                  version: Cl.buffer(Uint8Array.from([version])),
+                  hashbytes: Cl.buffer(Uint8Array.from(hashbytes)),
+                }),
+                Cl.uint(amount_ustx),
+                Cl.uint(first_rew_cycle),
+                Cl.uint(num_cycles),
+              ],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.ResponseErr));
+            assert(isClarityType(actual.value, ClarityType.Int));
+            expect(actual).toBeErr(Cl.int(expectedResponseErr));
+          }
+        )
+      );
+    });
+
+    it("should return (err 13) can-stack-stx for versions 0-4 pox addresses having hasbytes longer than 20", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.integer({ min: 0, max: 4 }),
+          fc.array(fc.nat({ max: 255 }), {
+            minLength: 21,
+            maxLength: 32,
```

Comment:
Are the array bounds really needed?

Reply:
The lower bound is needed because it is the specific tested case. The upper bound is needed because the clarity argument is (buff 32), and if the 32 length is exceeded the contract call will fail.